### ISSUE TITLE
native_posix: Warnings during link on orphan sections after #10368

### ISSUE THREAD

**Issue by pfalcon** [zephyrproject-rtos/zephyr#10493]
Resubmitted from https://github.com/zephyrproject-rtos/zephyr/pull/10368#issuecomment-428264786 :

After #10368 was applied, building for BOARD=native_posix produces warnings like below. The resulting executable seems to work.

~~~
[100%] Linking C executable zephyr.elf
/usr/bin/ld: warning: orphan section `.rel._k_mutex.static.pthread_key_lock' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o' being placed in section `.rel.dyn'.
/usr/bin/ld: warning: orphan section `.rel._k_mem_slab.static.posix_timer_slab' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o' being placed in section `.rel.dyn'.
/usr/bin/ld: warning: orphan section `.rel._k_sem.static.pthread_key_sem' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o' being placed in section `.rel.dyn'.
/usr/bin/ld: warning: orphan section `.rel._k_sem.static.mq_sem' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o' being placed in section `.rel.dyn'.
/usr/bin/ld: warning: orphan section `.rel.log_const_kernel' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o' being placed in section `.rel.dyn'.
Generating files from zephyr.elf for board: native_posix
~~~

**Comment by pfalcon** [comment]
cc: @aescolar

**Comment by aescolar** [comment]
Raising the priority to medium as this breaks sanitycheck in native_posix fully for "newer" OSs default installations.

**Comment by dcpleung** [comment]
GCC on Ubuntu has PIE turned on by default (since 16.10, IIRC). If you do gcc --v and there is an "--enable-default-pie", native_posix will fail to build. Adding "-no-pie" as linker option gets past the PIE issue, but it complains about not able to find "-lgcc_s" (libgcc_s.so and libgcc_s.so.1 are installed by GCC).

@aescolar Are you hitting the PIE issue? If so, we need to create another issue for that.

**Comment by aescolar** [comment]
> native_posix will fail to build

With "fail to build" you mean you will get a warning from the linker due to orphan sections?

Sidetrack: I did not try to hint about the PIE issue, but it is also an issue *1. It seems a bit dangerous to try to have this feature on by default in native_posix given that it is built with whatever the user wants/has locally.. 

1* `/usr/bin/x86_64-linux-gnu-ld: warning: orphan section `.rel.log_const_kernel' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib32/Scrt1.o' being placed in section `.rel.dyn'.`


> complains about not able to find "-lgcc_s" (libgcc_s.so and libgcc_s.so.1 are installed by GCC).

I did not see this one, even if I disable PIE with the compile+link options. But I guess I may have just installed more dependencies.

**Comment by dcpleung** [comment]
Not due to orphan sections, but something like this:

```
/usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../libapp.a(main.c.obj): warning: relocation in read-only section `.text.zephyr_app_main'
/usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: creating a DT_TEXTREL in object.
```

This causes sanitycheck to fail.

But PR #10528 will fix the orphan section warning emitted.

**Comment by aescolar** [comment]
Still some missing ones:
`
/usr/bin/x86_64-linux-gnu-ld: warning: orphan section `.rel.shell_cmd_' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib32/Scrt1.o' being placed in section `.rel.dyn'.
/usr/bin/x86_64-linux-gnu-ld: warning: orphan section `.rel.shell_module_' from `/usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib32/Scrt1.o' being placed in section `.rel.dyn'.
`

**Comment by dcpleung** [comment]
Hm... I haven't seen the shell ones on my development machine. It was on GCC 6.4.0 though.

The new PR #10536 should fix the shell_* section.
